src/library/Pagination/Pagination.js

```diff
@@ -0,0 +1,247 @@
+   * the current page on number entry
+   */
+  pageJumper?: boolean,
+  /** The number of items or data to be rendered on each page */
+  pageSize: number,
+  /**
+   * Render a [page sizer](#page-sizer)
+   * ([Select component](/components/select)); enables the user to select the
+   * page size
+   */
+  pageSizer?: boolean,
+  /**
+   * A collection of possible page sizes for a user to select from; implemented
+   * in [page sizer](#page-sizer). Be sure to provide unique values.
+   */
+  pageSizes?: Array<number>,
+  /** Available sizes */
+  size?: 'small' | 'medium' | 'large' | 'jumbo',
+  /** @Private */
+  theme: Object,
+  /** The total number of items or data to be paginated */
+  totalCount: number,
+  /**
+   * The number of consecutive pages to be displayed when the paginated results
+   * are truncated ([see example for details](#visible-range))
+   */
+  visibleRange?: number
+};
+
+export type Messages = {
+  pagination: { label: string, next: string, previous: string },
+  pageJumper?: { label: string, placeholder: string },
+  pageSizer?: {
+    status: (
+      category: string,
+      first: number,
+      last: number,
+      total: number
+    ) => string,
+    itemText: (pageSize: number) => string
+  }
+};
+
+export const componentTheme = (baseTheme: Object) => ({
+  PaginationPageJumper_width: 'auto',
+  Pagination_gutterWidth: baseTheme.space_inline_sm,
+  ...baseTheme
+});
+
+const styles = {
+  pageSizer: ({ theme }) => {
+    const rtl = theme.direction === 'rtl';
+    const middleMargin = rtl ? 'marginLeft' : 'marginRight';
+    return {
+      [middleMargin]: 'auto'
+    };
+  }
+};
+
+const Root = createStyledComponent(
+  Flex,
+  ({ theme: baseTheme }) => {
+    const theme = componentTheme(baseTheme);
+    return {
+      flexWrap: 'wrap-reverse',
+      marginBottom: `-${theme.Pagination_gutterWidth}`,
+
+      '& > *': {
+        marginBottom: theme.Pagination_gutterWidth
+      }
+    };
+  },
+  {
+    includeStyleReset: true,
+    withProps: {
+      element: 'nav',
+      justifyContent: 'end'
+    }
+  }
+);
+const PageSizer = createStyledComponent(_PageSizer, styles.pageSizer);
+
+/**
+ * Pagination offers a means to control the space consumed by a large data set
+ * or other collection of items by limiting the page size and providing
+ * navigation for access to all pages.
+ */
+class Pagination extends PureComponent<Props> {
+  static displayName = 'Pagination';
+  static defaultProps = {
+    category: 'rows',
+    messages: {
+      pagination: { label: 'Pagination', next: 'Next', previous: 'Previous' },
+      pageJumper: {
+        label: 'Jump to page',
+        placeholder: 'Page #'
+      },
+      pageSizer: {
+        status: (category, first, last, total) =>
+          `${first}–${last} of ${total} ${category}`,
+        itemText: (pageSize) => `${pageSize} per page`
+      }
+    },
+    pageSizes: [10, 20, 50],
```

Comment:
Would users ever need to specify an "all" option, in which case, this would need to accept an object of key/value pairs.  I would prolly vote for "not MVP".

Reply:
Yes, great point. I think we will save it for future work.